src/python/pants/backend/go/util_rules/external_module.py

```diff
@@ -42,6 +38,7 @@
 class DownloadExternalModuleRequest:
     path: str
     version: str
+    go_sum_digest: Digest = EMPTY_DIGEST
```

Comment:
In a followup, I'll remove this default and hook up the `ResolveExternalGoPackageRequest -> ResolvedGoPackage` rule to use `go_sum_digest`. Currently, it does not use it, so we try to use the generated `go.sum` - but that isn't being loaded properly and an integration test I wrote fails to build third-party packages.